dCommon/ZCompression.h

```diff
@@ -2,6 +2,8 @@
 
 #include <cstdint>
 
+#ifndef _WIN32
```

Comment:
These platform macros are sketchy at best and I think we should move away from using them. Maybe alongside these CMake changes we could introduce platform detection within CMake that defines our own macros (e.g. `DARKFLAME_PLATFORM_WINDOWS`, `DARKFLAME_PLATFORM_LINUX`, etc). CMake can much more reliably detect the platform where combinations of these macros will often fail due to compiler parity issues.

Reply:
If these weren't reliable, then I think every piece of c++ software would be having issues, no reason to fix what ain't broken

Reply:
> If these weren't reliable, then I think every piece of c++ software would be having issues, no reason to fix what ain't broken

I recall having issues with clang not working on the initial DLU release due to these macros being used. We also tend to use these platform macros as an analog for compiler (Windows == MSVC, Linux == GCC, etc). It really isn't that hard to implement detection in CMake and it looks nicer in code as well. Here's an example: https://github.com/codeshaunted/soul/blob/0286b126ff89d2e6ba5b285b631e6b1ece28d40f/CMakeLists.txt#L27

Reply:
Someone else can do this later if wanted, but I won't be.